apps/bluetooth/test/unit/views/pair_view_test.js

```diff
@@ -0,0 +1,456 @@
+    suiteTeardown(function() {
+      stubWindowClose.restore();
+      Pairview.pairButton.disabled = false;
+      Pairview.closeButton.disabled = false;
+      Pairview.pairDescription.textContent = undefined;
+    });
+
+    suite('click pair button > pair method: displaypasskey ', function() {
+      setup(function() {
+        this.sinon.stub(Pairview, '_pairMethod', 'displaypasskey');
+        Pairview.handleEvent({target: {id: 'button-pair'},
+                              type: 'click',
+                              preventDefault: fakePreventDefault});
+      });
+
+      test('description should be defined after clicked pair button',
+        function() {
+        assert.isDefined(Pairview.pairDescription.textContent);
+      });
+      test('pair button should be disabled after clicked event ', function() {
+        assert.isTrue(Pairview.pairButton.disabled);
+      });
+      test('close button should be disabled after clicked event ', function() {
+        assert.isTrue(Pairview.closeButton.disabled);
+      });
+      test('window should be closed after clicked event', function() {
+        assert.isTrue(stubWindowClose.called);
+      });
+    });
+
+    suite('click pair button > pair method: enterpincode ', function() {
+      var mockOptions, pincode = 'SixteenTxtLength';
+
+      setup(function() {
+        this.sinon.stub(Pairview, '_pairMethod', 'enterpincode');
+        Pairview.pinInput.value = pincode;
+        mockOptions = {
+          handle: {
+            setPinCode: function() {return Promise.resolve();}
+          }
+        };
+        this.sinon.spy(mockOptions.handle, 'setPinCode');
+        this.sinon.stub(Pairview, '_options', mockOptions);
+        Pairview.handleEvent({target: {id: 'button-pair'},
+                              type: 'click',
+                              preventDefault: fakePreventDefault});
+      });
+
+      teardown(function() {
+        Pairview.pinInput.value = '';
+      });
+
+      test('description should be defined after clicked pair button',
+        function() {
+        assert.isDefined(Pairview.pairDescription.textContent);
+      });
+      test('pair button should be disabled after clicked event ', function() {
+        assert.isTrue(Pairview.pairButton.disabled);
+      });
+      test('close button should be disabled after clicked event ', function() {
+        assert.isTrue(Pairview.closeButton.disabled);
+      });
+      test('set pin code should be set after clicked event', function() {
+        assert.isTrue(Pairview._options.handle.setPinCode.calledWith(pincode));
+      });
+      test('window should be closed after setPinCode resolve ', function(done) {
+        mockOptions.handle.setPinCode(pincode).then(() => {
+          assert.isTrue(stubWindowClose.called);
+        }, () => {
+          // reject case
+          assert.isTrue(false);
+        }).then(done, done);
+      });
+    });
+
+    suite('click pair button > pair method: confirmation, consent ',
```

Comment:
If the test case are for both `confirmation` and `consent`, we should test both methods. (Use an array loop to avoid duplicating the code)